staging/src/k8s.io/noderesourcetopology-api/pkg/apis/topology/v1alpha1/types.go

```diff
@@ -0,0 +1,148 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	_ "github.com/gogo/protobuf/gogoproto"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TopologyManagerPolicy represents kubelet configuration for
+// the TopologyManager component for a worker node.
+type TopologyManagerPolicy string
+
+const (
+	// SingleNUMANodeContainerLevel represents single-numa-node policy of
+	// the TopologyManager with container level scope
+	SingleNUMANodeContainerLevel TopologyManagerPolicy = "SingleNUMANodeContainerLevel"
+	// SingleNUMANodePodLevel represents single-numa-node policy of
+	// the TopologyManager with pod level scope
+	SingleNUMANodePodLevel TopologyManagerPolicy = "RestrictedPodLevel"
+	// RestrictedContainerLevel represent restricted policy of
+	// the TopologyManager with container level scope
+	RestrictedContainerLevel TopologyManagerPolicy = "RestrictedContainerLevel"
+	// RestrictedPodLevel represents restricted policy of
+	// the TopologyManager with pod level scope
+	RestrictedPodLevel TopologyManagerPolicy = "RestrictedPodLevel"
+	// BestEffortContainerLevel represent best-effort policy of
+	// the TopologyManager with container level scope
+	BestEffortContainerLevel TopologyManagerPolicy = "BestEffortContainerLevel"
+	// BestEffortPodLevel represents best-effort policy of
+	// the TopologyManager with pod level scope
+	BestEffortPodLevel TopologyManagerPolicy = "BestEffortPodLevel"
+	// None policy is the default policy and does not perform any topology alignment.
+	None TopologyManagerPolicy = "None"
+)
```

Comment:
I think these constants are related to implementation detail of particular resource policy implementation: TopologyManager + rest of bunch of managers inside Kubelet. I think those should be defined in other package. This API should be staying generic, without defining specifics of particular policy implementation details, as we don't want to update constants in two places: if TopologyManager introduces something new, we shouldn't be updating this API definitions.

Reply:
I see your point and our long term plan is indeed to remove these constants from the API definition. We intend to do that when the API to goes to `v1beta1` and have a few ideas to move in that direction. Since this is its first iteration of this API (proposed as `v1alpha1`), we would like components like NFD and NodeResourceTopology scheduler plugin to transition to it (from the previously used API https://github.com/k8stopologyawareschedwg/noderesourcetopology-api) as seamlessly as possible.

Reply:
Please refer to the ongoing discussion on this subject: https://github.com/k8stopologyawareschedwg/noderesourcetopology-api/issues/21. I expect outcome of this discussion to have an impact on how v1beta1 shapes up here.